### ISSUE TITLE
delete table then restart old parts resume

### ISSUE THREAD

**Issue by x923087131** [ClickHouse/ClickHouse#46942]
when i execute  delete sql
alter table fe_log  delete where time <='2022-10-25 11:17:40'

then query parts 

SELECT
count(1)
FROM system.parts
where table = 'fe_log'

has 47 parts;

then restart clickhouse i find  become 30000 parts and old part resume,about 10 mintue are deleted;

**Comment by x923087131** [comment]
engine is MergeTree

**Comment by Alex-Cheng** [comment]
```
SELECT
count(1)
FROM system.parts
where table = 'fe_log' and database = <your database> and is_active = 1
```

The parts which the column `is_active` is 0 will be ignored.

**Comment by x923087131** [comment]
i got it,i aim to remove inactive parts,  when i restart the inactive parts  will resume, about 10 mintue it will disappear in system.parts table and data path; each restart will consume  a lot of  memory and time althought active parts no much;

**Comment by Alex-Cheng** [comment]
Is your environment a cluster? Check your file system, when then inactive parts are removed the corresponding file directories should be removed. Thus it is not possible that the deleted parts come up again. I think probably there is something else that always resume the file directories. 

please check if the path of inactive parts still exist after these inactive parts are removed.

**Comment by x923087131** [comment]
> Is your environment a cluster? Check your file system, when then inactive parts are removed the corresponding file directories should be removed. Thus it is not possible that the deleted parts come up again. I think probably there is something else that always resume the file directories.
> 
> please check if the path of inactive parts still exist after these inactive parts are removed.


No,i just one  machine; now i found it is cause by wal.bin in table data path. Wal.bin file has lots of lod data and never been flushed that when i restart my machine  it will reading thd old data from wal.bin and then remove inactive data;
Now i am try to slove the question that how to flush wal.bin into device ;it can avoid what I said above;we could not close wal because we need it to imporve merge speed; 
Thanks for your aply.

Similar problems:
[[](url)](https://github.com/ClickHouse/ClickHouse/issues/17731)

**Comment by Algunenano** [comment]
> i got it,i aim to remove inactive parts, when i restart the inactive parts will resume, about 10 mintue it will disappear in system.parts table and data path;

By default CH deletes inactive parts after 8 minutes. See setting `old_parts_lifetime`.
